File: python/integration/runtime/send_email.py
from xml.etree.ElementTree import Element, tostring
import requests

# Transform json to xml
def json_to_xml(json_data):
    """
    :param json_data: json data
    :return: xml data
    """
    xml_data = Element('xml')
    for key, value in json_data.items():
        if isinstance(value, dict):
            xml_data.append(json_to_xml(value))
        else:
            xml_data.append(Element(key))
            xml_data[-1].text = str(value)
    return xml_data

#Send message via HTTP post to the external e-mail server
def post(url, data):
    try:
        r = requests.post(url, data=data)
        return r.text
    except requests.exceptions.RequestException as e:
        print(e)

# Function to convert JSON to XML
def json_to_xml(json_data):
    return(json2xml.Json2xml(json_data).to_xml())

#Send message via HTTP post to the external e-mail server
def post(url, data):
    try:
        r = requests.post(url, data=data)
        return r.text
    except requests.exceptions.RequestException as e:
        print(e)

def handler(event, context):
    xml = json_to_xml(event)
    xmlstr = tostring(xml, encoding='utf8', method='xml')
    print(xmlstr)
    return {
        "message": "E-mail sent"
    }

File: javascript/integration/runtime/send-email.js
var convert = require('xml-js')

// 1.) Convert JSON string to XML string
const jsonToXml = function (json) {
    return convert.json2xml(json, { compact: true, spaces: 4 })
}

// 2.) Send string with HTTP POST
const post = async (url, data) => {
    return new Promise((resolve, reject) => {
        const xhr = new XMLHttpRequest()
        xhr.open('POST', url)
        xhr.setRequestHeader('Content-Type', 'application/xml')
        xhr.onload = () => {
            if (xhr.status >= 200 && xhr.status < 300) {
                resolve(xhr.response)
            } else {
                reject(xhr.statusText)
            }
        }
        xhr.onerror = () => reject(xhr.statusText)
        xhr.send(data)
    })
}

exports.handler = async function (event, context) {

    // call method 1.) with var "event" to convert json to xml
    const data = jsonToXml(event)
    console.log(data)

    // call method 2.) to post xml
    // await post('https://www.example.com/sendmail', xml)

    return {
        statusCode: 200,
        message: "Success!"
    }
}
